yourls/Makefile

```diff
@@ -0,0 +1,106 @@
+
+.PHONY: shell
+shell:
+	@container=$$(eval "${BIN}/script-wizard choose 'docker exec -it into which container?' 'yourls' 'mysql' --default 'yourls'") && make --no-print-directory docker-compose-shell SERVICE=$${container}
+
+open-hook:
+	${BIN}/open /admin
+	
+.PHONY: list-admin-users # List admin users
+list-admin-users:
+# Pull the config.php file from the Docker container
+	@docker cp -a yourls-yourls-1:/var/www/html/user/config.php config.php.temp > /dev/null 2>&1
+	@echo "Existing admin users:"
+# Extract the usernames from $yourls_user_passwords array
+	@grep -oP "'\K[^']+(?=' *=>)" config.php.temp
+# Clean up temporary file
+	@rm config.php.temp
+	@echo
+
+.PHONY: add-admin-users # Add admin users
+add-admin-users:
+# Pull the config.php file from the Docker container
+	@docker cp -a yourls-yourls-1:/var/www/html/user/config.php config.php.temp > /dev/null 2>&1
+	@bash -c '\
+		while true; do \
+# Ask for username(s) and password(s) to add \
+			read -p "Enter username (leave blank to finish): " username; \
+			if [ -z "$$username" ]; then break; fi; \
+			read -p "Enter password: " password; \
+# Insert the new user into the $yourls_user_passwords array \
+			sed -i "/\$yourls_user_passwords\s*=/a \ \ '\''$$username'\'' => '\''$$password'\''," config.php.temp; \
+		done'
+# Copy the modified config.php back into the container
+	@docker cp -a config.php.temp yourls-yourls-1:/var/www/html/user/config.php > /dev/null 2>&1
+# Clean up temporary file
+	@rm config.php.temp
+	@echo
+	@echo 'Added the new admin users!'
+	@echo
+
+.PHONY: delete-admin-users # Delete admin users
+delete-admin-users:
+# Pull the config.php file from the Docker container
+	@docker cp -a yourls-yourls-1:/var/www/html/user/config.php config.php.temp > /dev/null 2>&1
+	@echo "Existing admin users:"
+# Extract the usernames from $yourls_user_passwords array
+	@grep -oP "'\K[^']+(?=' *=>)" config.php.temp
+	@while true; do \
+# Ask which user(s) to delete \
+		read -p "Enter the admin user(s) to delete (space-separated): " users; \
+		if [ -z "$$users" ]; then break; fi; \
+		for user in $$users; do \
+# Only delete username if it already exists \
+			if ! grep -P "'$$user' *=>" config.php.temp > /dev/null 2>&1; then \
+				echo "$$user is not an existing admin user."; \
+				users=$$(echo "$$users" | sed "s/\b$$user\b//g" | xargs); \
+				continue; \
+			fi; \
+# Delete the user(s) from the $yourls_user_passwords array \
+			sed -i "/'$$user' *=>/d" config.php.temp; \
+		done; \
+		echo "Deleted the admin users: $$users"; \
+		break; \
+	done
+# Copy the modified config.php back into the container
+	@docker cp -a config.php.temp yourls-yourls-1:/var/www/html/user/config.php > /dev/null 2>&1
+# Clean up temporary file
+	@rm config.php.temp
+	@echo
+
+.PHONY: add-plugins # Add plugins
+add-plugins:
+	@tar -C ${ROOT_DIR}/yourls/plugins -cf - . | docker exec -i yourls-yourls-1 tar -xf - -C /var/www/html/user/plugins/
```

Comment:
I think this should be baked into a Dockerfile so the plugins are part of the image.

Reply:
I don't think we should be distributing plugins in our repo, can they be downloaded as part of the Dockerfile, or simply a blank directory that the user must populate manually?

Reply:
So YOURLS comes with a few plugins installed by default. I wanted a few more installed by default when YOURLS is installed from d.ry, but I also wanted each user to decide for themselves which plugins to add. 

At the moment, I don't have a way to delete a plugin once you install it (you can _deactivate_ it from the UI, but you'd have to actually delete it manually in the container).

Do you still think that the set of plugins that _I_ want as default should be baked into the d.ry image? If so, should I remove the Makefile target to add more plugins, and just let the user do everything manually at that point?

Reply:
made notes in convo

Reply:
I think we should get rid of the add-plugins target, and build it into the Dockerfile instead.

Reply:
like in traefik for example:

https://github.com/EnigmaCurry/d.rymcg.tech/blob/4cd51673e2ae1ef3e7676ba6351b00cb656dbdd1/traefik/config/Dockerfile#L8C1-L8C41

https://github.com/EnigmaCurry/d.rymcg.tech/tree/master/traefik/config/context-template